jsonclient/client.go

```diff
@@ -228,7 +228,10 @@ func (c *JSONClient) PostAndParse(ctx context.Context, path string, req, rsp int
 		httpRsp.Body.Close()
 	}
 	if err != nil {
-		return nil, nil, RspError{StatusCode: httpRsp.StatusCode, Body: body, Err: err}
+		if httpRsp != nil {
```

Comment:
Yeah, the implementation has a code path that looks like it could return both.

Reply:
I see that there is a single documented case where it will return both non-nil:
https://github.com/golang/go/blob/bf86aec25972f3a100c3aa58a6abcbcc35bdea49/src/net/http/client.go#L478-L480

Reply:
Any chance we could get test coverage for this?

Reply:
I can't see an easy way to test this.